packages/cli-exec/src/exec.js

```diff
@@ -94,15 +94,41 @@ export const exec = command('exec', {
 
 // Spawn a command with cross-spawn and return an array containing the resulting status code along
 // with any error encountered while running. Uses a generator pattern to handle interupt signals.
-async function* spawn(cmd, args) {
+async function* spawn(cmd, args, percy) {
   let { default: crossSpawn } = await import('cross-spawn');
   let proc, closed, error;
 
   try {
-    proc = crossSpawn(cmd, args, { stdio: 'inherit' });
-    proc.on('close', code => (closed = code));
+    proc = crossSpawn(cmd, args, { stdio: 'pipe' });
+    // Writing stdout of proc to process
+    if (proc.stdout) {
+      proc.stdout.on('data', (data) => {
+        process.stdout.write(`${data}`);
+      });
+    }
+
+    if (proc.stderr) {
+      proc.stderr.on('data', (data) => {
+        process.stderr.write(`${data}`);
+      });
+    }
+
     proc.on('error', err => (error = err));
 
+    proc.on('close', code => {
+      closed = code;
+      if (code !== 0) {
+        // Only send event when there is a global error code
+        if (process.env.PERCY_TOKEN) {
```

Comment:
Can you add a comment around this condition?

Reply:
added a comment